docarray/doc_index/backends/elastic_doc_index.py

```diff
@@ -0,0 +1,484 @@
+                    row[col_name] = row[col_name] + 1.0e-9
+                request[col_name] = row[col_name]
+            requests.append(request)
+
+        _, warning_info = self._send_requests(requests, chunk_size)
+        for info in warning_info:
+            warnings.warn(str(info))
+
+        if refresh:
+            self._refresh(self._index_name)
+
+    def num_docs(self) -> int:
+        return self._client.count(index=self._index_name)['count']
+
+    def _del_items(
+        self,
+        doc_ids: Sequence[str],
+        chunk_size: Optional[int] = None,
+    ):
+        requests = []
+        for _id in doc_ids:
+            requests.append(
+                {'_op_type': 'delete', '_index': self._index_name, '_id': _id}
+            )
+
+        _, warning_info = self._send_requests(requests, chunk_size)
+
+        # raise warning if some ids are not found
+        if warning_info:
+            ids = [info['delete']['_id'] for info in warning_info]
+            warnings.warn(f'No document with id {ids} found')
+
+        self._refresh(self._index_name)
+
+    def _get_items(self, doc_ids: Sequence[str]) -> Sequence[TSchema]:
+        accumulated_docs = []
+        accumulated_docs_id_not_found = []
+
+        es_rows = self._client.mget(
+            index=self._index_name,
+            body={'ids': doc_ids},
+        )['docs']
+
+        for row in es_rows:
+            if row['found']:
+                doc_dict = row['_source']
+                accumulated_docs.append(doc_dict)
+            else:
+                accumulated_docs_id_not_found.append(row['_id'])
+
+        # raise warning if some ids are not found
+        if accumulated_docs_id_not_found:
+            warnings.warn(f'No document with id {accumulated_docs_id_not_found} found')
+
+        return accumulated_docs
+
+    def execute_query(self, query: Dict[str, Any], *args, **kwargs) -> Any:
+        if args or kwargs:
+            raise ValueError(
+                f'args and kwargs not supported for `execute_query` on {type(self)}'
+            )
+
+        resp = self._client.search(index=self._index_name, body=query)
+        docs, scores = self._format_response(resp)
+
+        return _FindResult(documents=docs, scores=np.array(scores))  # type: ignore
+
+    def _find(
+        self, query: np.ndarray, limit: int, search_field: str = ''
+    ) -> _FindResult:
+        if int(self._server_version.split('.')[0]) >= 8:
+            warnings.warn(
+                'You are using Elasticsearch 8.0+ and the current client is 7.10.1. HNSW based vector search is not supported and the find method has a default implementation.'
+            )
+
+        body = {
+            'size': limit,
+            'query': {
+                'script_score': {
+                    'query': {'match_all': {}},
+                    'script': {
+                        'source': f'cosineSimilarity(params.query_vector, \'{search_field}\') + 1.0',
```

Comment:
Should we make it possible for the user to select the similarity metric here and have a default set to `cosineSimilarity`?

Reply:
Here each `Similarity` has different script, for example `l2_norm` should be 
`'source': '1 / (1 + l2norm(params.query_vector, \'tens\'))'`
I think the conclusion of this discussion is we have a default implementation here, and user can use QueryBuilder if they want to pass other score standard.